src/components/Select/Select.tsx

```diff
@@ -0,0 +1,315 @@
+  const [hasValue, setHasValue] = useState<boolean>(!!(value || defaultValue));
+  const [overflowed, setOverflowed] = useState<boolean>(false);
+  const [titleText, setTitleText] = useState<string>("");
+
+  /**
+   * 내부 ref(overflowed 상태를 확인)와 외부에서 ref를 주입할 경우 덮어쓰기 방지를 위한 함수입니다.
+   * @param node - select 엘리먼트
+   */
+  const setRef = (node: HTMLSelectElement | null) => {
+    selectRef.current = node;
+    if (typeof ref === "function") {
+      ref(node);
+    } else if (ref) {
+      ref.current = node;
+    }
+  };
+
+  const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    if (onChange) onChange(e);
+    setHasValue(!!e.target.value && e.target.value !== "");
+    checkOverflow();
+  };
+
+  const checkOverflow = useCallback(() => {
+    const selectElement = selectRef.current;
+    if (!selectElement) return;
+
+    const selectedOption = selectElement.options[selectElement.selectedIndex];
+    const optionText = selectedOption ? selectedOption.text : "";
+
+    const computedStyle = window.getComputedStyle(selectElement);
+    const paddingLeft = parseFloat(computedStyle.paddingLeft) || 0;
+    const paddingRight = parseFloat(computedStyle.paddingRight) || 0;
+    const availableWidth =
+      selectElement.clientWidth - paddingLeft - paddingRight;
+
+    const canvas = canvasRef.current;
+    const context = canvas?.getContext("2d");
+    if (!context) return;
+
+    context.font = `${computedStyle.fontSize} ${computedStyle.fontFamily}`;
+    const textWidth = context.measureText(optionText).width;
+
+    const overflow = textWidth > availableWidth;
+    setOverflowed(overflow);
+    setTitleText(overflow ? optionText : "");
+  }, []);
+
+  const handleClear = () => {
+    const selectElement = selectRef.current;
+    if (disabled || !selectElement) return;
+
+    selectElement.value = "";
+    setHasValue(false);
+    if (onChange) {
+      const syntheticEvent = {
+        target: {
+          ...selectElement,
+          value: "",
+        },
+      } as React.ChangeEvent<HTMLSelectElement>;
+      onChange(syntheticEvent);
+    }
+    requestAnimationFrame(() => {
+      checkOverflow();
+    });
+  };
+
+  useEffect(() => {
+    canvasRef.current = document.createElement("canvas");
+  }, []);
+
+  useEffect(() => {
+    const rafId = requestAnimationFrame(() => {
+      checkOverflow();
+    });
+
+    const handleResize = () => {
+      checkOverflow();
+    };
+    window.addEventListener("resize", handleResize);
+    return () => {
+      cancelAnimationFrame(rafId);
+      window.removeEventListener("resize", handleResize);
+    };
+  }, [checkOverflow]);
+
+  const showClearButton = Boolean(clearButtonName && !disabled && hasValue);
+
+  return (
+    <div
+      data-tooltip={overflowed ? titleText : undefined}
+      className={wrapperStyles({ overflowed })}
+    >
+      <select
+        ref={setRef}
+        value={value}
+        defaultValue={value !== undefined ? undefined : defaultValue || ""}
```

Comment:
조건문은 부정으로 시작하기 보다 긍정형으로 하는 게 좀 더 직관적으로 이해하기 좋다고 생각해요 🤔

```suggestion
        defaultValue={value === undefined ?  : defaultValue || ""}
```

하지만 그럼에도 뒤에 OR 연산까지 있어서 개인적으로는 좀 더 이해하기 어려워보입니다...😂
저라면 의미 있는 변수를 사용해서 좀 더 읽기 쉽게 해볼 것 같은데 어떻게 생각하시나요?

```
const isControlled = value !== undefined;
defaultValue={isControlled ? undefined : (defaultValue || "")}
```

Reply:
말씀하신대로 복잡하여 가독성이 떨어지네요 별도의 변수로 분리하였습니다.
`isUncontrolled` `defaultValueStr`